### PR TITLE
ERA5 Native vertical grid zarr(`model-level-1h-0p25deg.zarr-v1`) added.

### PR DESCRIPTION
**Pull request by dabhicusp**
As mentioned in the 5th phase of the roadmap, I derived a high-resolution version of the ERA5 dataset(`gs://gcp-public-data-arco-era5/ar/model-level-1h-0p25deg.zarr-v1`). For deriving this dataset, I regridded the datasets to lat/long grids, converted the model level variables to pressure level variables, and computed some variables from the model level variables.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+RUN gcloud components install alpha --quiet
+
+COPY dist/arco_era5-0.1.0-py3-none-any.whl ./
+RUN pip install arco_era5-0.1.0-py3-none-any.whl
+
+# Copy files from official SDK image, including script/dependencies.
+COPY --from=beam_sdk /opt/apache/beam /opt/apache/beam
+
+# Set the entrypoint to Apache Beam SDK launcher.
+ENTRYPOINT ["/opt/apache/beam/boot"]
```
Comment (DarshanSP19) [inline Dockerfile:32]: nit: Add a new line. For all the relevant files.

File: src/arco_era5/__init__.py [hunk 2]
```diff
@@ -13,7 +13,7 @@
 # limitations under the License.
 from .data_availability import check_data_availability
 from .ingest_data_in_zarr import ingest_data_in_zarr_dataflow_job
-from .pangeo import run, parse_args
+# from .pangeo import run, parse_args
```
Comment (DarshanSP19) [inline src/arco_era5/__init__.py:16]: Really Needed?

File: src/model-level-native-vertical-zarr-initializer.py [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    hybrid = np.arange(1, 137 + 1, 1.0, dtype=np.float32)
+
+    coords = dict()
+    coords["time"] = pd.date_range( pd.Timestamp(start_date), pd.Timestamp(end_date),
+                                   freq=pd.DateOffset(hours=1), inclusive="left").values
+    coords['latitude'] = latitude
+    coords['longitude'] = longitude
+    coords['hybrid'] = hybrid
+
+    time_size = len(coords["time"])
```
Comment (DarshanSP19) [inline src/model-level-native-vertical-zarr-initializer.py:109]: Can use `time.size` and similarly for all the coordinates.

File: src/update-model-level-native-vertical-zarr-data.py [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+                        help='Start date, iso format string.')
+    parser.add_argument('-e', "--end_date", required=True,
+                        help='End date, iso format string.')
+    parser.add_argument("--init_date", type=str, default='1900-01-01',
+                        help="Date to initialize the zarr store.")
+
+    return parser.parse_known_args()
+
+known_args, pipeline_args = parse_arguments('')
+dates = hourly_dates(known_args.start_date, known_args.end_date)
```
Comment (DarshanSP19) [inline src/update-model-level-native-vertical-zarr-data.py:44]: Can use `daily_date_iterator` from `source_data.py` By adding a configurable `freq` parameter.

File: src/arco_era5/update_model_level_native_vertical_zarr.py [hunk 5]
```diff
@@ -0,0 +1,274 @@
...
+        wind_slice = ml_wind.sel(time=current_timestamp).compute()
+        moisture_slice = ml_moisture.sel(time=current_timestamp).compute()
+        surface_slice = sl_surface.sel(time=current_timestamp).compute()
+
+        wind_fieldset = mv.dataset_to_fieldset(self.attribute_fix(wind_slice).squeeze())
+        moisture_fieldset = mv.dataset_to_fieldset(self.attribute_fix(moisture_slice).squeeze())
+        surface_fieldset = mv.dataset_to_fieldset(self.attribute_fix(surface_slice).squeeze())
+
+        dataset = self.process_hourly_data([wind_fieldset, moisture_fieldset, surface_fieldset])
+        variables_full_names = {
```
Comment (DarshanSP19) [inline src/arco_era5/update_model_level_native_vertical_zarr.py:145]: Let's move this dict and URLs into a separate file.

File: src/arco_era5/update_model_level_native_vertical_zarr.py [hunk 6]
```diff
@@ -0,0 +1,274 @@
...
+            key (xb.Key): Offset dictionary for dimensions.
+            ds (xr.Dataset): Merged dataset for a single day.
+        """
+        logger.info("Inside the UpdateSlice function of the AR data.")
+        offset = key.offsets['time']
+        date = (datetime.datetime.strptime(self.init_date, '%Y-%m-%d') + 
+                datetime.timedelta(hours=offset))
+        zf = zarr.open(self.target)
+        region = slice(offset, offset + 1)
+        start_date = date.strftime('%Y-%m-%dT%H')
```
Comment (DarshanSP19) [inline src/arco_era5/update_model_level_native_vertical_zarr.py:243]: We can get rid of the start and end date. We can display `ds.time.values()` directly.

File: src/arco_era5/__init__.py (hunk 2 above)
Comment (dabhicusp) [reply src/arco_era5/__init__.py:16]: There are some libraries which are deprecated and still used inside the `.pangeo` files so need to comment this files.

**Review by DarshanSP19 — APPROVED** [review]

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,32 @@
+ARG py_version=3.9
```
Comment (mahrsee1997) [inline Dockerfile:1]: Add license string.

File: environment.yml [hunk 8]
```diff
@@ -0,0 +1,9 @@
+name: era5
```
Comment (mahrsee1997) [inline environment.yml:1]: Add license string.

File: src/arco_era5/__init__.py (hunk 2 above)
Comment (mahrsee1997) [reply src/arco_era5/__init__.py:17]: Lets discuss offline.

File: src/model-level-native-vertical-zarr-initializer.py [hunk 9]
```diff
@@ -0,0 +1,140 @@
...
+    output_path=known_args.output_path
+    start_date=known_args.start_date
+    end_date=known_args.end_date
+    init_date=known_args.init_date
+    from_init_date=known_args.from_init_date
+
+    zarr_chunks=(1, 18, 721, 1440)
+    template = make_template(init_date if from_init_date else start_date, end_date, zarr_chunks)
+    
+    print("Template created successfully.")
```
Comment (mahrsee1997) [inline src/model-level-native-vertical-zarr-initializer.py:135]: Use 'logging' instead of 'print'. Check at other places as well.

**Pull request merged**